File: Visitor/Javascript/CreditCard.js
import IPayment from "./IPayment.js";
import CPU from "./CPU.js";
import VGA from "./VGA.js";

class CreditCard extends IPayment {
  visit(component) {
    if (component instanceof CPU) {
      console.log("Charging credit card for CPU");
    } else if (component instanceof VGA) {
      console.log("Charging credit card for VGA");
    }
  }
}

export default CreditCard;

File: Visitor/Python 3/CreditCard.py
from IPayment import IPayment
from CPU import CPU
from VGA import VGA

class CreditCard(IPayment):
  def visit(self, component):
    if type(component) is CPU:
      print("Purchase CPU with Credit Card")
    elif type(component) is VGA:
      print("Purchase VGA with Credit Card")
